packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/methods/otp/services/auth_matrix_otp_service.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:widget_toolkit_otp/widget_toolkit_otp.dart';
+
+import '../../../models/auth_matrix_method_request.dart';
+import '../../../models/auth_matrix_response.dart';
+import '../../../repositories/auth_matrix_repository.dart';
+import '../models/auth_matrix_otp_payload.dart';
+
+class AuthMatrixOtpService extends SmsCodeService {
+  final AuthMatrixRepository _authMatrixRepository;
+  final AuthMatrixResponse _lastAuthMatrixResponse;
+
+  AuthMatrixOtpService({
+    required AuthMatrixRepository authMatrixRepository,
+    required AuthMatrixResponse lastAuthMatrixResponse,
+  })  : _authMatrixRepository = authMatrixRepository,
+        _lastAuthMatrixResponse = lastAuthMatrixResponse;
+
+  /// Confirms the phone code by the given [code] by sending it to the server.
+  ///
+  /// - [code] is the code that the user has received.
+  /// Returns a [Future] of [AuthMatrixResponse] that determines the next steps in the auth matrix process.
+  @override
+  Future confirmPhoneCode(String code) async =>
```

Comment:
I think we could specify the return type of this function, even tho it's a `Future<dynamic>`

Reply:
Done